plugins/modules/device_configs_backup_workflow_manager.py

```diff
@@ -514,109 +523,101 @@ def validate_site_exists(self, site_name):
         """
         site_exists = False
         site_id = None
+        response = None
 
-        # Attempt to retrieve site information from Catalyst Center
         try:
-            response = self.dnac._exec(
-                family="sites",
-                function="get_site",
-                op_modifies=True,
-                params={"name": site_name},
-            )
-            self.log("Response received post 'get_site' API call: {0}".format(str(response)), "DEBUG")
-            response = response.get("response")
-
-            # Process the response if available
-            if response:
-                site_id = response[0].get("id")
-                site_exists = True
-            else:
-                self.log("No response received from the 'get_site' API call.", "WARNING")
+            response = self.get_site(site_name)
+            if response is None:
+                raise ValueError
+            site = response.get("response")
+            site_id = site[0].get("id")
+            site_exists = True
 
         except Exception as e:
-            # Log an error message and fail if an exception occurs
-            self.log("An error occurred while retrieving site details for Site '{0}' using 'get_site' API call: {1}".format(site_name, str(e)), "ERROR")
-
-        # Update result if the site does not exist
-        if not site_exists:
-            self.msg = "An error occurred while retrieving site details for Site '{0}'. Please verify that the site exists.".format(site_name)
-            self.update_result("failed", False, self.msg, "ERROR")
+            self.status = "failed"
+            self.msg = ("An exception occurred: Site '{0}' does not exist in the Cisco Catalyst Center.".format(site_name))
+            self.result['response'] = self.msg
+            self.log(self.msg, "ERROR")
             self.check_return_status()
 
         return (site_exists, site_id)
 
     def get_device_ids_from_site(self, site_name, site_id):
```

Comment:
Move this API to dnac.py and make it common....

Reply:
Moved, I added two functions in dnac.py: one for getting device IDs from a site and another for retrieving the IP address from a device id.